test/e2e/performanceprofile/functests/utils/nodes/nodes.go

```diff
@@ -299,6 +300,37 @@ func GetNumaNodes(node *corev1.Node) (map[int][]int, error) {
 	return numaCpus, err
 }
 
+// GetCoreSiblings returns the siblings of core per numa node
+func GetCoreSiblings(node *corev1.Node) (map[int]map[int][]int, error) {
+	lscpuCmd := []string{"lscpu", "-e=node,core,cpu", "-J"}
+	out, err := ExecCommandOnNode(lscpuCmd, node)
+	var result NumaNodes
+	var numaNode, core, cpu int
+	coreSiblings := make(map[int]map[int][]int)
```

Comment:
Still thinking that we can simplify the map at least with types:
`type siblings []int
type cpuSiblings map[int]siblings
type nodeCpuSiblings map[int]cpuSiblings
`
`coreSiblings := make(nodeCpuSiblings)`
something like that

Reply:
I tried this but this looks like creating map in one steps we are doing it in multiple steps. Also one problem i hit using the above approach is, if i do like above, i don't know how to return coreSiblings because coresiblings is of type nodeCpuSiblings which is local to that function and is not a predeclared type like maps